### PR TITLE
Add conda disclaimer to docs

### PR DESCRIPTION
**Pull request by juntai-zheng**
## What changes are proposed in this pull request?
Adds a disclaimer regarding usage of conda in MLflow models.
image

It is similar to what has been added to the Databricks documentation of MLflow.
image

## How is this patch tested?

Manually tested.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [X] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -100,6 +100,31 @@ Additional Logged Files
...
+        - pip
+        - pip:
+            - mlflow
+            - scikit-learn==0.23.2
+            - cloudpickle==1.6.0
+              name: mlflow-env
+
+    If you would like to change the channel used in a model’s environment, you can re-register the model to the model registry with a new ``conda.yaml``. You can do this by specifying the channel in the ``conda_env`` parameter of ``log_model()``.
+
+    For more information on the ``log_model()`` API, see the MLflow documentation for the model flavor you are working with, for example, :py:func:`sklearn.log_model() <mlflow.sklearn.log_model>`.
```
Comment (harupy) [inline docs/source/models.rst:126]:
```suggestion
    For more information on the ``log_model()`` API, see the MLflow documentation for the model flavor you are working with, for example, :py:func:`mlflow.sklearn.log_model() <mlflow.sklearn.log_model>`.
```

Can we add `mlflow.`?

**Review by harupy — APPROVED** [review]
Left two minor comments, otherwise LGTM

**Review by dbczumar — APPROVED** [review]
LGTM!

File: docs/source/models.rst (hunk 1 above)
Comment (juntai-zheng) [reply docs/source/models.rst:126]: Soudns good!

File: docs/source/models.rst [hunk 2]
```diff
@@ -100,6 +100,31 @@ Additional Logged Files
...
+        channels:
+        - defaults
+        dependencies:
+        - python=3.8.8
+        - pip
+        - pip:
+            - mlflow
+            - scikit-learn==0.23.2
+            - cloudpickle==1.6.0
+      name: mlflow-env
```
Comment (harupy) [inline docs/source/models.rst:122]:
```suggestion
        dependencies:
        - python=3.8.8
        - pip
        - pip:
            - mlflow
            - scikit-learn==0.23.2
            - cloudpickle==1.6.0
        name: mlflow-env
```

It looks like [my code suggestion](https://github.com/mlflow/mlflow/pull/5554#discussion_r837962184) was incorrect.

Comment (dbczumar) [reply docs/source/models.rst:122]: Nit: `name` is generally listed at the top. Can we move it to the top?

Comment (harupy) [reply docs/source/models.rst:122]: +1

**Comment by harupy** [comment]
The updated code block looks ok:

image

image

**Pull request merged**